### DISCUSSION TITLE
how to make moon ci even quieter

### DISCUSSION THREAD

**Discussion by nhomble** [moonrepo/moon#1752] [Q&A] [answered]
# Background

We run `moon ci` in a single step/job in our CI pipeline. For certain changes that affect enough moon projects, we can end up running moon tasks on all of the projects.

# Problem

With enough of these projects, we end up going over our log size limit in our CI job window. This is  painful because if there is unit test error as part of the CI tasks, then we sometimes cannot see the feedback.

# Question

We see there is a 

>   -q, --quiet                      Hide all non-important terminal output [env: MOON_QUIET=]

parameter, but we are using that today. Maybe there is opportunity to hide even more information such as 

> [9:06:39 AM] Project '../../../../libs/foo/tsconfig.json' is up to date because newest input '../../../../libs/foo/src/bar.ts' is older than output '../../../../libs/foo/typings/tsconfig.tsbuildinfo'

**Comment by milesj** [comment, marked as answer]
@nhomble So this output is actually TypeScript:

> [9:06:39 AM] Project '../../../../libs/foo/tsconfig.json' is up to date because newest input '../../../../libs/foo/src/bar.ts' is older than output '../../../../libs/foo/typings/tsconfig.tsbuildinfo'

For that task, I believe you can remove `--verbose` from the `tsc` command.

With that said, all stdout/err output from tasks is printed to the terminal and cannot be turned off from moon arguments alone. However, you can set `MOON_OUTPUT_STYLE=none` to hide the output of all tasks, or configure `outputStyle` on a per-task basis: https://moonrepo.dev/docs/config/project#outputstyle
